### DISCUSSION TITLE
OTEL collector not available

### DISCUSSION THREAD

**Discussion by oscaratienzaatosnet** [open-telemetry/opentelemetry-java-instrumentation#7644] [Q&A]
can the agent be disabled when otel collector is not available? our app is not starting due to not collector available, i know about the -Dotel.javaagent.enabled=false but it would be nice if the instrumentation is disabled by its own when the collector is not reachable in X trys, else ... the app becomes unstable

**Comment by jkwatson** [comment]
Moving this discussion to the instrumentation project, where the agent is maintained.

**Comment by laurit** [comment]
Collector being unavailable should not prevent your application from starting. Please elaborate why do you believe it to be the case.

**Reply by oscaratienzaatosnet** [reply]
sorry for being late, i have started my app with consist in three microservices   curl----> A ----REST----> B ----GRPC---> C and i have started like this in docker:

ENV OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://my-release-tempo:4317
ENV OTEL_SERVICE_NAME=$APPNAME
ENV OTEL_TRACES_EXPORTER=otlp

ENTRYPOINT ["java","-javaagent:/opentelemetry-javaagent.jar","-jar","/APP.jar"] 

the agents by default says:

[otel.javaagent 2023-02-10 14:27:58:648 +0000] [OkHttp http://my-release-tempo:4317/...] ERROR io.opentelemetry.exporter.internal.grpc.OkHttpGrpcExporter - Failed to export spans. The request could not be executed. Full error message: my-release-tempo: Name or service not known
(this when tempo not deployed) 

and this when tempo not available:

023-02-10 15:30:48 [otel.javaagent 2023-02-10 14:30:48:146 +0000] [OkHttp http://my-release-tempo:4317/...] ERROR io.opentelemetry.exporter.internal.grpc.OkHttpGrpcExporter - Failed to export spans. The request could not be executed. Full error message: Failed to connect to my-release-tempo/127.0.0.1:4317
thought it was because tempo not in dns and put it in the etc hosts and result is the same

calls arent working. i can put more debug to traces ifg you can tell me how

**Reply by laurit** [reply]
These log lines just indicate that exporting spans fails. Could you explain what exactly is the issue here?

**Reply by oscaratienzaatosnet** [reply]
the main tx the one that works without opentelemetry agent is failing with the opentelemetry agent because it is not finding the server

**Reply by laurit** [reply]
Please describe how opentelemetry agent not being able to export spans can break your application or provide a minimal application that reproduces the issue you are observing.

**Reply by trask** [reply]
@oscaratienzaatosnet can you share the stack trace that shows your application failing to start? (not just the javaagent failing to send telemetry to the collector, which shouldn't affect your application since that's done in a background thread)
